### PR TITLE
Allow aux for convergence check

### LINKED ISSUE #19955: Allow the auxiliary solution to be used for steady state detection
## Reason
@aprilnovak proposed using auxiliary solutions to help detect steady-states for external problem solves

## Design
A few lines in `Transient.C` that allow the aux solution to be used in place of the nonlinear solution

## Impact
Make it easier to detect steady-states from external problem solves

### PR DESCRIPTION
**Pull request by lindsayad**
@aprilnovak can you test this out a bit and tell me whether it works for you?

Closes #19955

### CODE REVIEW

**Comment by aprilnovak** [comment]
I think something is still off - it does work as expected if I'm using `FEProblem`. But if using that `BasicExternalProblem` example I sent (I changed the value of the auxvariable to be `12345 * t` in order to get some time-dependence in there), it seems that the first time step still evaluates to NaN and then convergence is hit on the second step regardless of what I set the tolerance to:

```
Time Step 0, time = 0

Time Step 1, time = 1, dt = 1
Transfer FROM app, setting value of 12345
 Solve Converged!
Steady-State Relative Differential Norm: -nan

Time Step 2, time = 2, dt = 1
Transfer FROM app, setting value of 24690
 Solve Converged!
Steady-State Solution Achieved at time: 2
```

**Comment by lindsayad** [comment]
Ok, how about now? It looks better to me

**Comment by aprilnovak** [comment]
Works perfectly, thanks @lindsayad!

File: framework/src/executioners/Transient.C [hunk 1]
```diff
@@ -133,6 +133,10 @@ Transient::validParams()
   params.addParamNamesToGroup("start_time dtmin dtmax n_startup_steps trans_ss_check ss_check_tol "
                               "ss_tmin abort_on_solve_fail timestep_tolerance use_multiapp_dt",
                               "Advanced");
+  params.addParam<bool>("check_aux",
+                        false,
+                        "Whether to check the auxiliary system for convergence to steady-state. If "
+                        "false, then the nonlinear system is used.");
```
Comment (YaqiWang) [inline framework/src/executioners/Transient.C:139]: need to add this parameter in the Advanced group, or maybe add a new group for steady-state detection and put this parameter along with few others in the group.

File: test/src/problems/BasicExternalProblem.C [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#include "BasicExternalProblem.h"
+#include "AuxiliarySystem.h"
+
+registerMooseObject("MooseApp", BasicExternalProblem);
```
Comment (loganharbour) [inline test/src/problems/BasicExternalProblem.C:4]:
```suggestion
registerMooseObject("MooseTestApp", BasicExternalProblem);
```

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: framework/src/executioners/Transient.C (hunk 1 above)
Comment (lindsayad) [reply framework/src/executioners/Transient.C]: I went ahead and added the parameter to the existing "Steady State Detection" group

**Review by loganharbour — APPROVED** [review]

**Pull request merged**